client.go

```diff
@@ -133,11 +133,11 @@ type Client struct {
 // It is also possible to create a custom amqp.Config with whatever
 // configuration desired and that will be used as dial configuration when
 // connection to the message bus.
-func NewClient(url string) *Client {
+func NewClient(url string, dialConfig *DialConf) *Client {
```

Comment:
`Client` has a `WithTimeout(time.Duration)` function, perhaps we can set the dial timeout using a `WithDialTimeout(time.Duration)`?

Or if not, perhaps we can set it using a `WithDialConfig(DialConf)`?

I'm not a fan of having arguments that you mostly set as `nil` when using since most things in amqp-rpc has reasonable defaults.

Reply:
I second this. All the settings and configuration for both the client and the server uses the chained `With` pattern. Is there a reason we don't want to introduce `WithDialConfig`?

I don't mind having methods overriding others if they follow a clear pattern. In fact we already have that. `WithTLS` will configure TLS on the existing dialer so if you set it before `WithDialConfig` it will be overridden and if you set it after your TLS settings on the existing config will be overridden; see https://github.com/0x4b53/amqp-rpc/blob/7f5a4a292ef90bd300627e8e3e5132049a02ccae/client.go#L187-L192

Passing a timeout that just sets the timeout on the existing config feels better and let's us not have to pass `nil` everywhere.